### PR TITLE
Roboelectric Test Added

### PR DESCRIPTION
**Pull request by opticod**
@championswimmer , I have added the Roboelectric tests, I have also mocked the API , and from the API , inserted the data to database.This will test both the database insertion and HTTPfetching from APIs.

In previous tests , ```testDataInsertionIsCorrect()``` was not properly correct(GSON parsing was not properly done).

I have also fixed them in these tests.

This will also support issue #171 

Tests: 
- DbTest.java
- EventInsertTest.java
- MicrolocationInsertTest.java
- SessionInsertTest.java
- SpeakerInsertTest.java
- SponsorInsertTest.java
- TracksInsertTest.java

### CODE REVIEW

**Comment by opticod** [comment]
Also, made some necessary changes in ```build.gradle``` to make the build successful.
The build was initially failing due to a bug https://code.google.com/p/android/issues/detail?id=69174.
Here is the report of the test (runned in my machine) : 
screenshot from 2016-04-06 22-47-36

Please review.

**Comment by championswimmer** [comment]
LGTM

@mananwason take a look and merge when you can :)

**Comment by opticod** [comment]
@mananwason , can you please look into this

File: app/build.gradle [hunk 1]
```diff
@@ -102,3 +103,38 @@ dependencies {
     releaseCompile 'com.squareup.leakcanary:leakcanary-android-no-op:1.3.1' // or 1.4-beta1
     testCompile 'com.squareup.leakcanary:leakcanary-android-no-op:1.3.1' // or 1.4-beta1
 }
+
+jacoco {
```
Comment (championswimmer) [inline app/build.gradle]: all this is not needed btw, 

gradle has jacoco support.

I'll merge this nevertheless,

I'll do the unit test setup myself later.

**Pull request merged**